CMakeLists.txt

```diff
@@ -18,6 +18,20 @@ set(CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS true)
 # use our own version of FindBoost.cmake and other Find* scripts
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 
+# function to remove a flag from a variable
+function(RemoveFlag VAR SCOPE FLAG)
+	MESSAGE("Removing ${FLAG} flag from ${VAR}")
+	separate_arguments(${VAR})
+	list(REMOVE_ITEM ${VAR} ${FLAG})
+	string(REPLACE ";" " " ${VAR} "${${VAR}}")
```

Comment:
I'm not sure if the indirection is needed for the input. Try

```cmake
string(REPLACE ";" " " ${VAR} ${VAR})
```

Reply:
I tested to be sure, but it is necessary to do it that way, like how the `check_compiler_has_flag` macro does it.